core/controllers/topics_and_skills_dashboard.py

```diff
@@ -79,6 +80,21 @@ def get(self):
                             self.user, topic_rights)
                     )
 
+        all_classroom_dict = config_domain.TOPIC_IDS_FOR_CLASSROOM_PAGES.value
+        classroom_names = [
+            classroom['name'] for classroom in all_classroom_dict]
+
+        for topic_summary_dict in topic_summary_dicts:
+            classroom_is_assigned_to_topic = False
+            for classroom in all_classroom_dict:
+                if topic_summary_dict['id'] in classroom['topic_ids']:
+                    classroom_is_assigned_to_topic = True
+                    topic_summary_dict['classroom'] = classroom['name']
+                    break
+
+            if not classroom_is_assigned_to_topic:
+                topic_summary_dict['classroom'] = None
```

Comment:
You can just do this before the inner for loop, and then you don't need the extra boolean variable at all.

Reply:
Woah! that's smart. Done.